### PR TITLE
DAOS-10854 test: verify multi-user dfuse permissions

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: dfuse_mu
Skip-unit-tests: true
Skip-fault-injection-test: true
   
- Add ExecutableCommand.run_user
  - Used to run a command as another user
  - Update dfuse to run commands as another user
- Add ExecutableCommand.bind_cores
  - Used to run a command with taskset
  - Update dfuse to use this option
- Add utility verify_perms.py to verify owner, group, other permissions
  on a file or directory.
  - Add VerifyPermsCommand wrapper to run remotely
- Add dfuse/mu_perms.py to verify multi-user permissions
- Adjust dfuse/mu_mount.py to use updated test utilities
    
Required-githooks: true
    
Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
@ashleypittman Mostly FYI for the progress of this. The real workhorse is `verify_perms.py`, which can be called outside of a test case. When more test cases are added, I'll look at expanding that utility.

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/util/verify_perms.py [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+        raise ValueError(f'Path already exists: {path}')
+    logger.info('Creating %s %s', entry_type, path)
+    if owner:
+        owner_uid_gid = get_user_uid_gid(owner)
+        _as_user(owner_uid_gid, _create, path, entry_type)
+    else:
+        _create(path, entry_type)
+
+
+def _create(path, entry_type):
```
Comment (shimizukko) [inline src/tests/ftest/util/verify_perms.py:56]: You might want to add method description.

File: src/tests/ftest/util/verify_perms.py [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+        return
+    logger.info('Removing %s', path)
+    if owner:
+        owner_uid_gid = get_user_uid_gid(owner)
+        _as_user(owner_uid_gid, _delete, path)
+    else:
+        _delete(path)
+
+
+def _delete(path):
```
[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/tests/ftest/util/verify_perms.py:92]: Done, thanks

File: src/tests/ftest/util/verify_perms.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/verify_perms.py:56]: Done, thanks

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Comment by daltonbohning** [comment]
Latest test run passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10885/14/testReport/
With tags: `dfuse_mu pr,vm dfusedaosbuild dfuse,pr dfuse,daily_regression`

**Pull request merged**